scripts/run_integration_tests.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+# Usage - run commands from repo root:
+# To check if new changes to the library cause changes to any snapshots:
+#   ./scripts/run_integration_tests.sh
+# To regenerate snapshots:
+#   UPDATE_SNAPSHOTS=true ./scripts/run_integration_tests.sh
+
+set -e
+
+# To add new tests create a new ts file in the 'integration_tests/stacks' directory, append its file name to the STACK_CONFIGS array.
+# Note: Each ts file will have its respective snapshot built in the snapshots directory, e.g. lambda-function-stack.ts
+#       will generate both snapshots/test-lambda-function-stack-snapshot.json and snapshots/correct-lambda-function-stack-snapshot.json
+STACK_CONFIGS=("lambda-function-stack" "lambda-nodejs-function-stack")
+
+script_path=${BASH_SOURCE[0]}
+scripts_dir=$(dirname $script_path)
+repo_dir=$(dirname $scripts_dir)
+root_dir=$(pwd)
+if [[ "$root_dir" =~ .*"datadog-cdk-constructs/scripts".* ]]; then
+    echo "Make sure to run this script from the root $(datadog-cdk-constructs) directory, aborting"
+    exit 1
+fi
+
+integration_tests_dir="$repo_dir/integration_tests/"
+if [ "$UPDATE_SNAPSHOTS" = "true" ]; then
+    echo "Overwriting snapshots in this execution"
+fi
+
+# Login to ECR. This is a required step in order to pull a public Docker image for the PythonFunction construct
+aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws
+
+yarn
+yarn build
+
+cd $integration_tests_dir
+
+OUTPUT_ARRAY+=("====================================")
+allTestsPassed=true
+compose_output() {
+     if [ $1 -eq 0 ]; then
+        OUTPUT_ARRAY+=("SUCCESS: There were no differences between the $2 and $3")
+    else
+        allTestsPassed=false
+        OUTPUT_ARRAY+=("FAILURE: There were differences between the $2 and $3. Review the diff output above.")
+        OUTPUT_ARRAY+=("If you expected the $2 to be different generate new snapshots by running this command from a development branch on your local repository: 'UPDATE_SNAPSHOTS=true ./scripts/run_integration_tests.sh'")
+    fi
+    OUTPUT_ARRAY+=("====================================")
+}
```

Comment:
I think this is a good improvement, right now the output looks like this:

```
====================================
  asset-output/index.js  249b 
Performing diff of snapshots/test-lambda-function-stack-snapshot.json against snapshots/correct-lambda-function-stack-snapshot.json

SUCCESS: There were no differences between the snapshots/test-lambda-function-stack-snapshot.json and snapshots/correct-lambda-function-stack-snapshot.json
====================================
Performing diff of snapshots/test-lambda-nodejs-function-stack-snapshot.json against snapshots/correct-lambda-nodejs-function-stack-snapshot.json
SUCCESS: There were no differences between the snapshots/test-lambda-nodejs-function-stack-snapshot.json and snapshots/correct-lambda-nodejs-function-stack-snapshot.json
====================================
```

I wonder if we can improve this further and make it look more like the output of our unit tests:

```
PASS: test-lambda-function-stack
PASS: test-lambda-nodejs-function-stack

SUCCESS: 2 out of 2 snapshot tests passed
```

Also, maybe we can log the test results individually as they are completed, then log the overall result at the end? Rather than adding to the output and printing the entire output at the end.

Reply:
A couple of reasons for this:
- More readable output = faster debugging
- Logging the results live = easier to see the progress of the tests while they run